Add data from migrations table to report

sharidas opened the pull request:
Add data from migrations table to the
report.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

sharidas commented:
Fixes https://github.com/owncloud/configreport/issues/63

patrickjahns commented:
Would be great to expand the test-matrix now, to test configreport app in combination with several Database backends. ( mysql / postgres / oracle ) - so we can ensure that the report still works expected with the different supported databases

PVince81 commented:
wot

```
phpdbg -qrr -d memory_limit=4096M -d zend.enable_gc=0 "../../lib/composer/bin/phpunit" --configuration ./tests/unit/phpunit.xml
PHPUnit 5.7.27 by Sebastian Bergmann and contributors.

Runtime:       PHPDBG 7.1.20-1+ubuntu16.04.1+deb.sury.org+1
Configuration: /var/www/owncloud/apps/configreport/tests/unit/phpunit.xml

..Makefile:41: recipe for target 'test-php-unit-dbg' failed
```

https://drone.owncloud.com/owncloud/configreport/121/64

@patrickjahns

sharidas commented on tests/unit/ReportDataCollectorTest.php:
```diff
@@ -0,0 +1,86 @@
+			->willReturn($expectedResult);
+
+		$this->connection
+			->method('getQueryBuilder')
```
@PVince81 I captured the data from my dev machine and updated the test. Let me know if the updated test looks ok. For testing I used sqlite. Below is the console log:

```
➜  unit git:(add-oc-migration-data) ✗ /home/sujith/test/owncloud3/lib/composer/phpunit/phpunit/phpunit ReportDataCollectorTest.php
Cannot load Xdebug - it was already loaded
PHPUnit 5.7.27 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.2.9-1+ubuntu18.04.1+deb.sury.org+1 with Xdebug 2.6.0
Configuration: /home/sujith/test/owncloud3/apps/configreport/tests/unit/phpunit.xml

.                                                                   1 / 1 (100%)

Time: 10.24 seconds, Memory: 44.00MB

OK (1 test, 181 assertions)

Generating code coverage report in Clover XML format ... done
➜  unit git:(add-oc-migration-data) ✗ 
```

sharidas replied:
Ouch hold on... I will ping again here. Just hold on for review. Until I fix this...

sharidas replied:
Ok, so the failure as of now is `..Makefile:41: recipe for target 'test-php-unit-dbg' failed` https://drone.owncloud.com/owncloud/configreport/125/64.

@PVince81 Let me know the updated tests look ok.

PVince81 commented on tests/unit/ReportDataCollectorTest.php:
```diff
@@ -0,0 +1,82 @@
+			$this->assertArrayHasKey('app', $result);
+			$this->assertArrayHasKey('version', $result);
+			$values = \array_values($result);
+			$this->assertContains($values[0], ['core', 'dav', 'files_sharing', 'files_trashbin', 'notifications', 'dav', 'files_external', 'federatedfilesharing', 'customgroups']);
```
this will likely fail as we don't bundle "customgroups", etc

only pick the ones present in the core repository

sharidas replied:
I have updated the test, removed `customgroups`. I can't delete `notifications`, though. As the bundle `owncloud-daily-master-qa.tar.bz2` has notifications app.

PVince81 commented:
@sharidas you cannot list all migrations anyway, so only a few is enough.
As for apps, you also used `assertContains()` which is not an exact match but only a few, so removing "notifications" is ok.

sharidas commented:
Updated the test. Removed notification from the list. The iteration is done over an array of  results, and if an app not part of the core app is found then no assertions are checked. If the apps are from core the assertions are checked.

PVince81 commented:
@patrickjahns should we ignore / disable to unblock this ?

```
+ if [ -z "true" ]; then make test-php-unit; fi
+ if [ "true" = "true" ]; then make test-php-unit-dbg; fi
phpdbg -qrr -d memory_limit=4096M -d zend.enable_gc=0 "../../lib/composer/bin/phpunit" --configuration ./tests/unit/phpunit.xml
PHPUnit 5.7.27 by Sebastian Bergmann and contributors.

Runtime:       PHPDBG 7.1.20-1+ubuntu16.04.1+deb.sury.org+1
Configuration: /var/www/owncloud/apps/configreport/tests/unit/phpunit.xml

make: *** [test-php-unit-dbg] Error 255
..Makefile:41: recipe for target 'test-php-unit-dbg' failed
```

not sure if related to this PR

sharidas commented:
> Code looks good 

I am going to push the change for backport then. Thanks.

sharidas commented:
Backport to stable10 branch https://github.com/owncloud/configreport/pull/68

patrickjahns commented:
> not sure if related to this PR

Before this PR, the debugger would not cause segfaults and run out of memory. 4G should be limited to 2G anyway

PVince81 commented:
Any solutions in scope for 10.0.10 RC2 ?

sharidas commented:
The pr is updated by skipping the test.
